api/v1alpha3/azuremachine_types.go

```diff
@@ -44,6 +44,12 @@ type AzureMachineSpec struct {
 	// +optional
 	Image *Image `json:"image,omitempty"`
 
+	// Identity is the type of identity used for the virtual machine.
+	// The type 'SystemAssigned' is an implicitly created identity
```

Comment:
Please add a blurb about the role the identity is assigned. Subscription contributor is pretty significant. We'd want users to understand the role being granted to the VM.

Reply:
is there somewhere else to update in the docs as well?

Reply:
a small blurb in https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/master/docs/topics about identity would be 💯

Reply:
you could even reuse what you and David wrote in https://docs.google.com/document/d/17o3uDJmsYlzfaC00K0dzYOaYin04zew853VYYxW2vhc/edit?usp=sharing for the concepts